lib/ansible_tower_client/collection.rb

```diff
@@ -11,19 +11,19 @@ def all
     end
 
     def collection_for(paginated_result)
-      body = JSON.parse(paginated_result.body)
```

Comment:
I switched everything to YAML.safe_load for consistency since safe_load handles both types.  Though you are right - it is always going to be JSON.  If you feel that JSON.parse would be better for JSON only responses - I can make that change.